lldb/include/lldb/Target/Target.h

```diff
@@ -622,6 +623,30 @@ class Target : public std::enable_shared_from_this<Target>,
   ///     requirements.
   llvm::Error SetLabel(llvm::StringRef label);
 
+  /// Get the target session name for this target.
+  ///
+  /// Provides a meaningful name for IDEs or tools to display for dynamically
+  /// created targets. Defaults to "Session {ID}" based on the globally unique
+  /// ID.
+  ///
+  /// \return
+  ///     The target session name for this target.
+  const std::string &GetTargetSessionName() { return m_target_session_name; }
```

Comment:
```suggestion
  llvm::StringRef GetTargetSessionName() { return m_target_session_name; }
```

You should use StringRef. It can represent different kinds of strings automatically.